### PR TITLE
feat: add `fig-autocomplete` subcommand

### LINKED ISSUE #1118: Shell completion improvements
- Add Fig.io integration: https://fig.io/docs/guides/autocomplete-for-teams
- Could `$SHELL` be used to automatically choose the shell when not specified? If not, then running `infracost completion` should show the help text as well as the existing error message. In the same way that `infracost diff` does.

### PR DESCRIPTION
**Pull request by fedeci**
This does not completely close (#1118), but I am almost done with creating a GH action that will allow to automatically push new versions of the spec to the autocomplete repo and I will open another PR soon!

### CODE REVIEW

**Comment by aliscott** [comment]
Thanks @fedeci this looks great! I think it needs `make test_update_cmd` run to update the command golden file tests to fix the build.

**Comment by fedeci** [comment]
Sure!

File: cmd/infracost/testdata/help/help.golden [hunk 1]
```diff
@@ -18,14 +18,14 @@ EXAMPLES
...
 AVAILABLE COMMANDS
-  breakdown   Show full breakdown of costs
-  comment     Post an Infracost comment to GitHub, GitLab, Azure Repos or Bitbucket
-  completion  Generate shell completion script
-  configure   Display or change global configuration
-  diff        Show diff of monthly costs between current and planned state
-  help        Help about any command
-  output      Combine and output Infracost JSON files in different formats
-  register    Register for a free Infracost API key
+  breakdown        Show full breakdown of costs
```
Comment (aliscott) [inline cmd/infracost/testdata/help/help.golden:21]: Any idea why this changes? Is it a cobra bug that still considers hidden commands when indenting?

**Review by aliscott — APPROVED** [review]
LGTM, I had a play around with this on fig and it works well!

One small question but it's not an issue for merging.

File: cmd/infracost/testdata/help/help.golden (hunk 1 above)
Comment (fedeci) [reply cmd/infracost/testdata/help/help.golden:21]: I tried to figured out that too, apparently it is a cobra bug.

Comment (aliscott) [reply cmd/infracost/testdata/help/help.golden:21]: Yep, looks like a cobra bug.
